src/loaders/ncbi_downloader/ncbi_downloader.py

```diff
@@ -0,0 +1,82 @@
+"""
+Download genome files from NCBI FTP server
+"""
+
+import os
+import time
+from datetime import datetime
+
+import ftputil
+import pytz
+
+
+def _download_genome_file(download_dir: str, gene_id: str, target_file_ext: list[str]) -> None:
+    # NCBI file structure: a delegated directory is used to store files for all versions of a genome
+    # e.g. File structure for RS_GCF_000968435.2 (https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/968/435/)
+    # genomes/all/GCF/000/968/435/ --> GCF_000968435.1_ASM96843v1/
+    #                              --> GCF_000968435.2_ASM96843v2/
+    ncbi_domain, user_name, password = 'ftp.ncbi.nlm.nih.gov', 'anonymous', 'anonymous@domain.com'
+
+    success, attempts, max_attempts, gene_id = False, 0, 3, gene_id[3:]
```

Comment:
Will all gene IDs, including non-refseq gene IDs, have a 3 char prefix?

Reply:
yup. All 300K from GTDB metadata files have that 3-char prefix.

Reply:
Ok, I didn't really grok that this was GTDB specific. If the GTDB 3 char prefix doesn't exist in other sources we might have to make mods here but we can cross that bridge when we come to it